### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by Lagrang3**
This PR consists of code cleanup.

### CODE REVIEW

File: include/boost/math/fft/dft_api.hpp [hunk 1]
```diff
@@ -398,11 +445,16 @@
...
-      executor_r2c::execute(in_first,in_last,out,
-        [this](const real_type* i, complex_type* o)
+      
+      std::vector<real_type,allocator_t> hc(size(),alloc);
+      
+      executor_halfcomplex::execute(in_first,in_last,hc.data(),
+        [this](const real_type* i, real_type* o)
         {
-          backend::template real_to_complex<complex_type>(i,o);
+          backend::real_to_halfcomplex(i,o);
```
Comment (Lagrang3) [inline include/boost/math/fft/dft_api.hpp:461]: On the real transforms, is natural to work from R^N space to 1/2 C^N space and back. Any more degrees of freedom on the complex side would be spurious. Hence my rfft api is built based on R^N -> 1/2 C^N. The complex type in the output is deduced from the iterator traits. This is related to issue #14.

**Comment by ckormanyos** [comment]
Hi @Lagrang3 I could not solve this particular problem in the configuration available.

I was stumped by this problem and will address it with clear advice.

I reverted a few changes and got CI running again. I found that you/we had either now or sooner (not sure when) added even more link dependencies such as `pmr` to some test files. I advise against adding these sorts of link dependencies, as we would like CI to rely on as little as possible.

Also, I encourage us to be wise and avoid breaking CI so close to the deadline. I can accept a few tweaks and changes to push forward and improve. But leaving a hanging branch with broken CI does not make us look very strong and good. It leaves the wrong impression about the quality of our/your work. Appearance does matter a bit at this stage. In summary, I strongly encourage clean branches, absolutely a clean develop branch running CI and low risk taking prior to deadline.

You have done great work  --- keep it that way.

@cosurgi

**Comment by Lagrang3** [comment]
@ckormanyos. I understand. In fact, to me the changes seemed quite innocent and some where bug fixes. 
In any case PR are not compulsory to merge. You're right we are too close to the deadline. We could stop at this point either by merging this or not. And in the future we will finish the unfinished.

**Comment by ckormanyos** [comment]
>  We could stop at this point either by merging this or not. And in the future we will finish the unfinished.

Good choice Eduardo. OK @Lagrang3 I've merged and we will proceed according to your words...

**Pull request merged**